cmd/zdb/zdb.c

```diff
@@ -8490,17 +8490,17 @@ zdb_read_block(char *thing, spa_t *spa)
 	    !(flags & ZDB_FLAG_DECOMPRESS)) {
 		const blkptr_t *b = (const blkptr_t *)(void *)
 		    ((uintptr_t)buf + (uintptr_t)blkptr_offset);
-		if (zfs_blkptr_verify(spa, b, B_FALSE, BLK_VERIFY_ONLY) ==
-		    B_FALSE) {
+		if (zfs_blkptr_verify(spa, b,
+		    BLK_CONFIG_NEEDED, BLK_VERIFY_ONLY) == B_FALSE) {
```

Comment:
You'll get a zfs_dbgmsg here now where you would not have before.  Doesn't hurt anything but IIRC it might happen quite a bit since indirect blocks are compressed and the blkptr looks "wrong" until decompressed.

Reply:
From reading the code, I think the impact would be that if you do `zdb -G -R ...:i` / `:b` (but not `:d`), then when it prints the dbgmsg there might be _one_ additional "bad" block pointer printed, when it tries to misinterpret the compressed data as a block pointer.  That seems fine.